File: lodomaniak-mobile-ui/src/app/services/follow.service.ts
import { Injectable } from "@angular/core";
import { ApiService } from "../core/api.service";
import { Observable } from "rxjs/Observable";
import { map } from "rxjs/operators";
import { Follow } from "../model/follow.model";
import { HttpParams } from "@angular/common/http";

export const MY_FOLLOW_PATH = '/follow/mine';
export const FOLLOW_PATH = '/follow';

@Injectable()
export class FollowService {

  constructor(private apiService: ApiService) {
  }

  getMyFollow(followType: string, followedObjectId: number) {
    return this.apiService.get(MY_FOLLOW_PATH, new HttpParams({
      fromObject: {
        followType: followType,
        followedObjectId: followedObjectId.toString(),
      }
    }));
  }

  addFollow(follow: Follow): Observable<Follow> {
    return this.apiService.post(FOLLOW_PATH, follow)
      .pipe(map((response) => (response.body)));
  }

  removeFollow(followId: number): Observable<any> {
    return this.apiService.delete(FOLLOW_PATH + '/' + followId.toString());
  }

}

File: lodomaniak-follow/src/main/java/pl/lodomaniak/follow/FollowService.java
package pl.lodomaniak.follow;

import org.springframework.security.core.userdetails.User;
import pl.lodomaniak.follow.api.FollowTO;
import pl.lodomaniak.follow.api.FollowType;
import pl.lodomaniak.follow.spi.FollowExternalService;
import pl.lodomaniak.user.api.exception.UserNotFoundException;

public interface FollowService extends FollowExternalService {

    FollowTO getMyFollow(Long followedObjectId, FollowType followType, User user) throws UserNotFoundException;

    FollowTO addFollow(FollowTO follow, User user) throws UserNotFoundException;

    void removeFollow(Long followId, User user) throws UserNotFoundException;

}
